pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,206 @@
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("proxyconfig-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// We only care about a configmap source with a specific name/namespace,
+	// so filter events before they are provided to the controller event handlers.
+	pred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return handleConfigMap(e.MetaNew)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return handleConfigMap(e.Meta)
+		},
+	}
+
+	// Watch for changes to the additional trust bundle configmap.
+	if err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred); err != nil {
+		return err
+	}
+
+	// Watch for changes to the proxy resource.
+	if err = c.Watch(&source.Kind{Type: &configv1.Proxy{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// handleConfigMap returns true if meta namespace is "openshift-config".
+// "openshift-config" is the namespace for one or more
+// ConfigMaps that contain user provided trusted CA bundles.
+func handleConfigMap(meta metav1.Object) bool {
+	return meta.GetNamespace() == "openshift-config"
+}
+
+var _ reconcile.Reconciler = &ReconcileProxyConfig{}
+
+// ReconcileProxyConfig reconciles a ClusterLogging object
+type ReconcileProxyConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func proxyNamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Name: "cluster",
+	}
+}
+
+// Reconcile reads that state of the cluster for a cluslter-scoped
+// named "cluster" or a configmap object in namespace "openshift-config".
+func (r *ReconcileProxyConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.NamespacedName == proxyNamespacedName() {
+		proxyConfig := &configv1.Proxy{}
+		logrus.Debugf("Proxyconfig reconcile request.Name: '%s'", request.Name)
+		if err := r.client.Get(context.TODO(), request.NamespacedName, proxyConfig); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				logrus.Errorf("proxy not found; reconciliation will be skipped - request %v", request)
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, fmt.Errorf("failed to get proxy '%s': %v", request.Name, err)
```

Comment:
this doesn't requeue the request...

Reply:
Oops.   I thought by returning non-nil error, it'd be requeued...  Or you mean we should _not_ requeue here?  (I vaguely remember you mentioned we should not requeue..., but not certain.  sorry...)
```
// The Controller will requeue the Request to be processed again if the returned error is non-nil or
// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
```